code/modules/overmap/exoplanets/decor/_turfs.dm

```diff
@@ -17,20 +17,32 @@
 	var/has_edge_icon = TRUE
 
 /turf/simulated/floor/exoplanet/New()
-	// try to get the the atmos and area of the exoplanet
+	// try to get the the atmos and area of the planet
 	if(SSatlas.current_map.use_overmap)
-		var/obj/effect/overmap/visitable/sector/exoplanet/E = GLOB.map_sectors["[z]"]
-		if(istype(E))
-			if(E.atmosphere)
-				initial_gas = E.atmosphere.gas.Copy()
-				temperature = E.atmosphere.temperature
+		// if exoplanet
+		var/obj/effect/overmap/visitable/sector/exoplanet/exoplanet = GLOB.map_sectors["[z]"]
+		if(istype(exoplanet))
+			if(exoplanet.atmosphere)
+				initial_gas = exoplanet.atmosphere.gas.Copy()
+				temperature = exoplanet.atmosphere.temperature
 			else
 				initial_gas = list()
 				temperature = T0C
 			//Must be done here, as light data is not fully carried over by ChangeTurf (but overlays are).
-			set_light(MINIMUM_USEFUL_LIGHT_RANGE, E.lightlevel, E.lightcolor)
-			if(E.planetary_area && istype(loc, world.area))
-				ChangeArea(src, E.planetary_area)
+			set_light(MINIMUM_USEFUL_LIGHT_RANGE, exoplanet.lightlevel, exoplanet.lightcolor)
+			if(exoplanet.planetary_area && istype(loc, world.area))
+				ChangeArea(src, exoplanet.planetary_area)
+		// if away site
+		var/datum/map_template/ruin/away_site/away_site = GLOB.map_templates["[z]"]
+		if(istype(away_site))
+			if(away_site.exoplanet_atmosphere)
+				initial_gas = away_site.exoplanet_atmosphere.gas.Copy()
+				temperature = away_site.exoplanet_atmosphere.temperature
+			else
+				initial_gas = list()
+				temperature = T0C
+			set_light(MINIMUM_USEFUL_LIGHT_RANGE, away_site.lightlevel, away_site.lightcolor)
```

Comment:
This is repeating the code above, it should instead be made to work without repetition.
Also, this is a double check, if one is an exoplanet it's not an away_site and vice versa.
While this would normally be admissible, this thing runs five billion times (once per turf essentially, on every exoplanet turf) and should be as efficient as possible, this is a proc where performance matters.

Reply:
i think it's better to not separate the stuff here into a proc because of proc overhead reasons

the other thing was addressed